libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessageParser.java

```diff
@@ -124,29 +124,8 @@ public boolean parse(ByteBuffer buffer) throws IOException {
 
             position = buffer.position();
 
-            if (checkSumEnabled) {
-                buffer.position(position + length);
-
-                // Garbled message
-                tag = FIXTags.get(buffer);
-                if (tag == 0)
-                    continue;
-
-                // Garbled message
-                if (!checkSum.get(buffer))
-                    continue;
-
-                // Garbled message
-                if (tag != CheckSum)
-                    continue;
-
-                // Garbled message
-                if ((FIXCheckSums.sum(buffer, beginning, position - beginning + length) & 0xff)
-                        != checkSum.asInt())
-                    continue;
-
-                buffer.position(position);
-            }
+            if (checkSumEnabled && !acceptCheckSum(buffer, beginning, position, length))
```

Comment:
is it "accept" or maybe "validate"? 🤷‍♂

Reply:
I think "accept" reads better when a method returns a Boolean value and "validate" when the acceptance or rejection is signaled some other way. 🙂

Reply:
Correct, true wordsmith!